src/utils/alert.ts

```diff
@@ -72,34 +72,62 @@ export const validateResponse = (
   return checks
 }
 
-export const getMessageForAlert = (
-  alert: string,
-  url: string,
-  ipAddress: string,
+export const getMessageForAlert = ({
+  alert,
+  url,
+  ipAddress,
+  status,
+  trueThreshold,
+}: {
+  alert: string
+  url: string
+  ipAddress: string
   status: string
-): {
+  trueThreshold: number
+}): {
   subject: string
   body: string
 } => {
   const getSubject = (url: string, status: string) => {
-    if (status === 'UP') {
-      return `${alert} from probing ${url} has been resolved.`
+    if (alert === 'status-not-2xx') {
+      const statusAlert = `Target ${url} is not OK`
+      if (status === 'UP') {
+        return `[RECOVERY] ${statusAlert}`
+      }
+      return `[INCIDENT] ${statusAlert}`
+    }
+
+    const responseAlert = `Target ${url} takes long to respond`
+    if (alert === 'response-time-greater-than-400-ms' && status === 'UP') {
+      return `[RECOVERY] ${responseAlert}`
     }
-    return `"${alert}" has been detected from probing ${url}`
+    return `[INCIDENT] ${responseAlert}`
   }
 
   const getBody = (status: string) => {
-    if (status === 'UP') {
-      return `We found that your "${alert}" alert has been resolved when probing "${url}".`
+    if (status === 'DOWN') {
+      if (alert === 'status-not-2xx') {
+        return `Target ${url} is not healthy. It has not been returning status code 2xx ${trueThreshold} times in a row.`
+      }
+
+      if (alert === 'response-time-greater-than-400-ms') {
+        return `Target ${url} is not healthy. The response time has been greater than 400 ms ${trueThreshold} times in a row`
+      }
+
+      return `Target ${url} is not healthy.`
     }
-    return `New "${alert}" alert when probing "${url}".`
+
+    return `Target ${url} is back to healthy.`
   }
 
+  const now = Date.now()
+  const today = new Date(now)
```

Comment:
Can we refactore line 123, 124, and 130 to be `const now = new Date().toUTCString()`?

![image](https://user-images.githubusercontent.com/19143644/112273411-e68b8200-8caf-11eb-9ed5-ca138e1f6d26.png)

Reply:
fixed